enderio-base/src/main/java/com/enderio/base/common/handler/travel/TeleportHandler.java

```diff
@@ -0,0 +1,150 @@
+package com.enderio.base.common.handler.travel;
+
+import com.enderio.base.common.init.EIOItems;
+import com.enderio.base.common.item.darksteel.IDarkSteelItem;
+import com.enderio.base.config.base.BaseConfig;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.EntityTeleportEvent;
+
+import javax.annotation.Nullable;
+import java.util.Comparator;
+import java.util.Optional;
+
+/**
+ * Thanks to the developers of <a href="https://github.com/castcrafter/travel_anchors">https://github.com/castcrafter/travel_anchors</a> for allowing us to use their code with our license.
+ * For their agreements look at doc/license/castcrafter/travel-anchors/license.md in the repo root
+ */
+
+public class TeleportHandler {
+
+    public static boolean canTeleport(Player player) {
+        return canTeleport(player, InteractionHand.MAIN_HAND) || canTeleport(player, InteractionHand.OFF_HAND);
+    }
+
+    public static boolean canTeleport(Player player, InteractionHand hand) {
+        ItemStack stack = player.getItemInHand(hand);
+        if (stack.getItem() == EIOItems.TRAVEL_STAFF.get())
+            return true;
+        if (stack.getItem() instanceof IDarkSteelItem darkSteelItem) {
+            //TODO: Check for upgrade;
+        }
```

Comment:
The ITravelTarget interface can be used as an optional dependency, because it doesn't need to be implemented on minecraft exposed registry objects like item and blocks. Forcing an interface on Items or Blocks (what ctm is doing rn), would make our mod to be a required dependency. This method is only called for particle rendering and other clients can add their own particles or use our addParticleMethod, if we make it public

Reply:
Sounds good